runtime/common/src/migrations/liquidity_pools_gateway.rs

```diff
@@ -0,0 +1,131 @@
+use cfg_types::domain_address::Domain;
+use frame_support::{
+	pallet_prelude::ValueQuery,
+	storage::{child::MultiRemovalResults, generator::StorageValue, unhashed},
+	storage_alias,
+	traits::{Get, OnRuntimeUpgrade},
+	Blake2_128Concat, StorageMap,
+};
+use pallet_liquidity_pools_gateway::{pallet::Pallet as LPGateway, Config};
+use pallet_order_book::weights::Weight;
+use sp_runtime::{DispatchError, TryRuntimeError};
+use sp_std::vec::Vec;
+
+#[storage_alias]
+pub type OutboundMessageNonceStore<T: Config> =
+	StorageValue<pallet_liquidity_pools_gateway::Pallet<T>, u64, ValueQuery>;
+
+#[storage_alias]
+pub type OutboundMessageQueue<T: Config + frame_system::Config> = StorageMap<
+	pallet_liquidity_pools_gateway::Pallet<T>,
+	Blake2_128Concat,
+	u64,
+	(
+		Domain,
+		<T as frame_system::Config>::AccountId,
+		pallet_liquidity_pools::Message,
+	),
+>;
+
+#[storage_alias]
+pub type FailedOutboundMessages<T: Config + frame_system::Config> = StorageMap<
+	pallet_liquidity_pools_gateway::Pallet<T>,
+	Blake2_128Concat,
+	u64,
+	(
+		Domain,
+		<T as frame_system::Config>::AccountId,
+		pallet_liquidity_pools::Message,
+		DispatchError,
+	),
+>;
+
+const LOG_PREFIX: &str = "LPGatewayMessageQueue";
+
+pub struct Migration<T>(sp_std::marker::PhantomData<T>);
+
+impl<T> OnRuntimeUpgrade for Migration<T>
+where
+	T: Config + frame_system::Config,
+{
+	fn on_runtime_upgrade() -> Weight {
+		let mut weight = Self::clear_storage(
+			OutboundMessageNonceStore::<T>::storage_prefix(),
+			"OutboundMessageNonceStore",
+		);
+
+		for (nonce, entry) in OutboundMessageQueue::<T>::iter() {
+			log::warn!("{LOG_PREFIX}: Found outbound message:\nnonce:{nonce}\nentry:{entry:?}");
+			weight = weight.saturating_add(<T as frame_system::Config>::DbWeight::get().reads(1));
+		}
+
+		for (nonce, entry) in FailedOutboundMessages::<T>::iter() {
+			log::warn!(
+				"{LOG_PREFIX}: Found failed outbound message:\nnonce:{nonce}\nentry:{entry:?}"
+			);
+			weight = weight.saturating_add(<T as frame_system::Config>::DbWeight::get().reads(1));
+		}
```

Comment:
I'm pretty sure if they're empty, we do not need to remove them. 

cc @wischli

Reply:
Yea, this is more like an extra check to ensure that they're empty. Not doing any specific removals here apart from the nonce storage.